kroxylicious-api/src/main/java/io/kroxylicious/proxy/config/tls/KeyPairSet.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config.tls;
+
+import java.util.List;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Represents a set of KeyPairs, from which we can select one for each VirtualCluster
+ * that has SANs containing the bootstrap host.
+ * @param keyPairs
+ */
+public record KeyPairSet(@JsonProperty(required = true) List<KeyPair> keyPairs) implements KeyProvider {
+
+    @Override
+    public <T> T accept(KeyProviderVisitor<T> visitor) {
+        return visitor.visit(this);
+    }
+
+    // we want to obtain a key pair where all target hostnames match a SAN, literal or wildcard in the certificate
```

Comment:
Do we want to insist that all there's a single certificate per virtual cluster?   The user could issue separate certificates - one for bootstrap, one for each broker.  I've no idea how likely/unlikely this possibility is.

Reply:
Good point. I saw Strimzi are introducing certificate-per-broker https://github.com/strimzi/strimzi-kafka-operator/issues/7687. I guess this lets you be more precise, so you can be sure you are talking to broker-x, not accidentally to another broker offering a shared cert. Maybe some users would want a similar setup for the Proxy.

Wonder if this means we would need to take the list of certificates and rather than picking one for the vcluster, we'd have to build up a KeyStore. (so the feature becomes more about KeyStore construction from a set of PEMs)

Reply:
Yes, I'm wondering if ultimately io.kroxylicious.proxy.model.VirtualCluster#getDownstreamSslContext should pass the sniHostname, allow the VirtualCluster to return an SSLContext configured with the most appropriate certification.  

However, I think assuming a single certificate per virtual cluster is a valid starting simplification.  We can enhance later.